subprojects/core-api/src/main/java/org/gradle/api/artifacts/ResolutionStrategy.java

```diff
@@ -366,6 +367,29 @@
      */
     ResolutionStrategy dependencySubstitution(Action<? super DependencySubstitutions> action);
 
+    /**
+     * Gradle implicitly registers dependency substitution rules for all configurations in the whole build
+     * tree to find projects in other included builds. These rules are always active by default.
+     *
+     * There are however cases, where a certain configuration should not apply these rules when resolving.
+     * For example, if a binary version of a module should be discovered that is also represented by
+     * a project in another build.
+     *
+     * This deactivates these global substitution rules.
+     *
+     * @since 7.4
+     */
+    @Incubating
+    ResolutionStrategy disableGlobalDependencySubstitutionRules();
```

Comment:
There should be a method to query the state of this flag to help observe the state for logic or reporting.

Reply:
Should it be:

```
Property<Boolean> getUseGlobalDependencySubstitutionRules()
```
?

Reply:
I tried to follow the existing pattern of method naming `ResolutionStrategy` (which is probably not so good...).

`Property<Boolean> getUseGlobalDependencySubstitutionRules()` would be an option to have only one method for everything. If the use of `Property` is desired here. Otherwise a getter+setter would be an option to have two methods (instead of three).

Reply:
Let's go with a `Property` indeed

Reply:
@ljacomet I turned this into a property as suggested. Maybe you can run this PR through the CI pipeline to make sure it does not break anything unexpectedly.